src/conversion.rs

```diff
@@ -502,6 +587,68 @@ impl IntoPy<Py<PyTuple>> for () {
     fn into_py(self, py: Python<'_>) -> Py<PyTuple> {
         PyTuple::empty(py).unbind()
     }
+
+    #[inline]
+    fn __py_call_vectorcall1<'py>(
+        self,
+        py: Python<'py>,
+        function: Borrowed<'_, 'py, PyAny>,
+    ) -> PyResult<Bound<'py, PyAny>> {
+        unsafe {
+            cfg_if::cfg_if! {
+                if #[cfg(all(
+                    not(PyPy),
+                    not(GraalPy),
+                    any(Py_3_10, all(not(Py_LIMITED_API), Py_3_9)) // PyObject_CallNoArgs was added to python in 3.9 but to limited API in 3.10
+                ))] {
+                    // Optimized path on python 3.9+
+                    ffi::PyObject_CallNoArgs(function.as_ptr()).assume_owned_or_err(py)
+                } else {
+                    ffi::PyObject_Call(function.as_ptr(), PyTuple::empty(py).as_ptr(), std::ptr::null_mut()).assume_owned_or_err(py)
+                }
+            }
+        }
+    }
+
+    #[inline]
+    fn __py_call_vectorcall<'py>(
+        self,
+        py: Python<'py>,
+        function: Borrowed<'_, 'py, PyAny>,
+        kwargs: Option<Borrowed<'_, '_, PyDict>>,
+    ) -> PyResult<Bound<'py, PyAny>> {
+        match kwargs {
+            Some(kwargs) => unsafe {
+                ffi::PyObject_Call(
+                    function.as_ptr(),
+                    PyTuple::empty(py).as_ptr(),
+                    kwargs.as_ptr(),
+                )
+                .assume_owned_or_err(py)
+            },
+            None => <Self as IntoPy<Py<PyTuple>>>::__py_call_vectorcall1(self, py, function),
```

Comment:
e.g. if `PyObject_CallNoArgs` was in `ffi::compat`, I think this could be (IMO clearer):

```suggestion
            None => unsafe { ffi::compat::PyObject_CallNoArgs(function.as_ptr()).assume_owned_or_err(py) },
```

Reply:
Please rebase and use the symbol I now added in #4461 😄